nameko_grpc/streams.py

```diff
@@ -166,7 +166,7 @@ def headers_to_send(self, defer_until_data=True):
         if self.headers_sent or len(self.headers) == 0:
             return False
 
-        if defer_until_data and self.queue.empty():
+        if defer_until_data and self.queue.empty() and self.buffer.empty():
```

Comment:
if flush_to_buffer is called then the queue may be empty while data sits in the buffer.